modules/auxiliary/scanner/discovery/ipv6_neighbor_router_advertisement.rb

```diff
@@ -33,20 +33,22 @@ def initialize
       OptInt.new('TIMEOUT_NEIGHBOR', [true, "Time (seconds) to listen for a solicitation response.", 1])
     ], self.class)
 
-    register_advanced_options(
-      [
-        OptString.new('PREFIX', [true, "Prefix that each host should get an IPv6 address from",
-          "2001:1234:DEAD:BEEF::"]
-        )
-      ], self.class)
-
     deregister_options('SNAPLEN', 'FILTER', 'RHOST', 'PCAPFILE')
   end
 
+  def generate_prefix()
+    max = 16 ** 4
+    prefix = "2001::"
```

Comment:
an ipv6 address cannot have more than one double-colon sequence. I'll fix it up.